xbmc/peripherals/bus/PeripheralBus.h

```diff
@@ -54,7 +54,7 @@ namespace PERIPHERALS
     /*!
      * @return True if this bus needs to be polled for changes, false if this bus performs updates via callbacks
      */
-    bool NeedsPolling(void) const { return m_bNeedsPolling; }
+    bool NeedsPolling(void) const { CSingleLock lock(m_critSection); return m_bNeedsPolling; }
```

Comment:
See `Initialise` for instance. You have to do CAS to make that right, so I'm not sure if it's worth it. It's perfectly fine to locks here as long as you dont block during any external calls.

Right now there is some members are unprotected (because they are read-only), there's the event which is thread safe, some use the critical section, dont forget the subclasses that have access to all of them, and adding atomics to the equation... What do you say? Even worth adding more complexity to it?